// @react-navigation/material-top-tabs/lib/commonjs/views/MaterialTopTabBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = TabBarTop;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _reactNativeTabView = require("react-native-tab-view");

var _native = require("@react-navigation/native");

var _color = _interopRequireDefault(require("color"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function TabBarTop(props) {
  const {
    colors
  } = (0, _native.useTheme)();

  const {
    state,
    navigation,
    descriptors,
    activeTintColor = colors.text,
    inactiveTintColor = (0, _color.default)(activeTintColor).alpha(0.5).rgb().string(),
    allowFontScaling = true,
    showIcon = false,
    showLabel = true,
    pressColor = (0, _color.default)(activeTintColor).alpha(0.08).rgb().string(),
    iconStyle,
    labelStyle,
    indicatorStyle,
    style
  } = props,
        rest = _objectWithoutProperties(props, ["state", "navigation", "descriptors", "activeTintColor", "inactiveTintColor", "allowFontScaling", "showIcon", "showLabel", "pressColor", "iconStyle", "labelStyle", "indicatorStyle", "style"]);

  return React.createElement(_reactNativeTabView.TabBar, _extends({}, rest, {
    navigationState: state,
    activeColor: activeTintColor,
    inactiveColor: inactiveTintColor,
    indicatorStyle: [{
      backgroundColor: colors.primary
    }, indicatorStyle],
    style: [{
      backgroundColor: colors.card
    }, style],
    pressColor: pressColor,
    getAccessibilityLabel: (_ref) => {
      let {
        route
      } = _ref;
      return descriptors[route.key].options.tabBarAccessibilityLabel;
    },
    getTestID: (_ref2) => {
      let {
        route
      } = _ref2;
      return descriptors[route.key].options.tabBarTestID;
    },
    onTabPress: (_ref3) => {
      let {
        route,
        preventDefault
      } = _ref3;
      const event = navigation.emit({
        type: 'tabPress',
        target: route.key,
        canPreventDefault: true
      });

      if (event.defaultPrevented) {
        preventDefault();
      }
    },
    onTabLongPress: (_ref4) => {
      let {
        route
      } = _ref4;
      return navigation.emit({
        type: 'tabLongPress',
        target: route.key
      });
    },
    renderIcon: (_ref5) => {
      let {
        route,
        focused,
        color
      } = _ref5;

      if (showIcon === false) {
        return null;
      }

      const {
        options
      } = descriptors[route.key];

      if (options.tabBarIcon !== undefined) {
        const icon = options.tabBarIcon({
          focused,
          color
        });
        return React.createElement(_reactNative.View, {
          style: [styles.icon, iconStyle]
        }, icon);
      }

      return null;
    },
    renderLabel: (_ref6) => {
      let {
        route,
        focused,
        color
      } = _ref6;

      if (showLabel === false) {
        return null;
      }

      const {
        options
      } = descriptors[route.key];
      const label = options.tabBarLabel !== undefined ? options.tabBarLabel : options.title !== undefined ? options.title : route.name;

      if (typeof label === 'string') {
        return React.createElement(_reactNative.Text, {
          style: [styles.label, {
            color
          }, labelStyle],
          allowFontScaling: allowFontScaling
        }, label);
      }

      return label({
        focused,
        color
      });
    }
  }));
}

const styles = _reactNative.StyleSheet.create({
  icon: {
    height: 24,
    width: 24
  },
  label: {
    textAlign: 'center',
    textTransform: 'uppercase',
    fontSize: 13,
    margin: 4,
    backgroundColor: 'transparent'
  }
});
