// @react-navigation/material-top-tabs/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "createMaterialTopTabNavigator", {
  enumerable: true,
  get: function get() {
    return _createMaterialTopTabNavigator.default;
  }
});
Object.defineProperty(exports, "MaterialTopTabView", {
  enumerable: true,
  get: function get() {
    return _MaterialTopTabView.default;
  }
});
Object.defineProperty(exports, "MaterialTopTabBar", {
  enumerable: true,
  get: function get() {
    return _MaterialTopTabBar.default;
  }
});
Object.defineProperty(exports, "MaterialTopTabNavigationOptions", {
  enumerable: true,
  get: function get() {
    return _types.MaterialTopTabNavigationOptions;
  }
});
Object.defineProperty(exports, "MaterialTopTabNavigationProp", {
  enumerable: true,
  get: function get() {
    return _types.MaterialTopTabNavigationProp;
  }
});
Object.defineProperty(exports, "MaterialTopTabBarProps", {
  enumerable: true,
  get: function get() {
    return _types.MaterialTopTabBarProps;
  }
});
Object.defineProperty(exports, "MaterialTopTabBarOptions", {
  enumerable: true,
  get: function get() {
    return _types.MaterialTopTabBarOptions;
  }
});

var _createMaterialTopTabNavigator = _interopRequireDefault(require("./navigators/createMaterialTopTabNavigator"));

var _MaterialTopTabView = _interopRequireDefault(require("./views/MaterialTopTabView"));

var _MaterialTopTabBar = _interopRequireDefault(require("./views/MaterialTopTabBar"));

var _types = require("./types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
