// @react-navigation/material-bottom-tabs/lib/module/views/MaterialBottomTabView.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import * as React from 'react';
import { StyleSheet } from 'react-native';
import { BottomNavigation, DefaultTheme, DarkTheme } from 'react-native-paper';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import { TabActions, useTheme } from '@react-navigation/native';
export default function MaterialBottomTabView(_ref) {
  let {
    state,
    navigation,
    descriptors
  } = _ref,
      rest = _objectWithoutProperties(_ref, ["state", "navigation", "descriptors"]);

  const {
    dark,
    colors
  } = useTheme();
  const theme = React.useMemo(() => {
    const t = dark ? DarkTheme : DefaultTheme;
    return _objectSpread({}, t, {
      colors: _objectSpread({}, t.colors, {}, colors, {
        surface: colors.card
      })
    });
  }, [colors, dark]);
  return React.createElement(BottomNavigation, _extends({}, rest, {
    theme: theme,
    navigationState: state,
    onIndexChange: index => navigation.dispatch(_objectSpread({}, TabActions.jumpTo(state.routes[index].name), {
      target: state.key
    })),
    renderScene: (_ref2) => {
      let {
        route
      } = _ref2;
      return descriptors[route.key].render();
    },
    renderIcon: (_ref3) => {
      let {
        route,
        focused,
        color
      } = _ref3;
      const {
        options
      } = descriptors[route.key];

      if (typeof options.tabBarIcon === 'string') {
        return React.createElement(MaterialCommunityIcons, {
          name: options.tabBarIcon,
          color: color,
          size: 24,
          style: styles.icon,
          importantForAccessibility: "no-hide-descendants",
          accessibilityElementsHidden: true
        });
      }

      if (typeof options.tabBarIcon === 'function') {
        return options.tabBarIcon({
          focused,
          color
        });
      }

      return null;
    },
    getLabelText: (_ref4) => {
      let {
        route
      } = _ref4;
      const {
        options
      } = descriptors[route.key];
      return options.tabBarLabel !== undefined ? options.tabBarLabel : options.title !== undefined ? options.title : route.name;
    },
    getColor: (_ref5) => {
      let {
        route
      } = _ref5;
      return descriptors[route.key].options.tabBarColor;
    },
    getBadge: (_ref6) => {
      let {
        route
      } = _ref6;
      return descriptors[route.key].options.tabBarBadge;
    },
    getAccessibilityLabel: (_ref7) => {
      let {
        route
      } = _ref7;
      return descriptors[route.key].options.tabBarAccessibilityLabel;
    },
    getTestID: (_ref8) => {
      let {
        route
      } = _ref8;
      return descriptors[route.key].options.tabBarTestID;
    },
    onTabPress: (_ref9) => {
      let {
        route,
        preventDefault
      } = _ref9;
      const event = navigation.emit({
        type: 'tabPress',
        target: route.key,
        canPreventDefault: true
      });

      if (event.defaultPrevented) {
        preventDefault();
      }
    }
  }));
}
const styles = StyleSheet.create({
  icon: {
    backgroundColor: 'transparent'
  }
});
